### PR TITLE
Allow using several custom template directories

### PR DESCRIPTION
**Pull request by babolivier**
Allow using several directories in `read_templates`. Useful for #10596, and also allowing modules to specify their own module directory _on top_ of the configured default one(s).

### CODE REVIEW

File: synapse/config/_base.py [hunk 1]
```diff
@@ -262,18 +262,23 @@ def read_templates(
...
+                    if not self.path_exists(custom_template_directory):
+                        raise ConfigError(
+                            "Configured template directory does not exist: %s"
+                            % (custom_template_directory,)
+                        )
 
-            # Search the custom template directory as well
-            search_directories.insert(0, custom_template_directory)
+                    # Search the custom template directory as well
+                    search_directories.insert(0, custom_template_directory)
```
Comment (richvdh) [inline synapse/config/_base.py:281]: this means we'll search the directories in reverse order to that given in the arguments, I think? Is that deliberate? It should probably be documented if so? If not, it suggests we need more tests?

Comment (babolivier) [reply synapse/config/_base.py]: It was not deliberate, thanks for pointing it out! I've added a test that checks the order we search directories in.

File: tests/config/test_base.py [hunk 2]
```diff
@@ -74,8 +74,66 @@ def test_loading_custom_templates(self):
...
+        """Tests that directories are searched in the right order if multiple custom
+        template directories are provided.
+        """
+        # Create two temporary directories on the filesystem.
+        tempdirs = [
+            tempfile.TemporaryDirectory(),
+            tempfile.TemporaryDirectory(),
+        ]
+
+        # Create one template in each directory, which content is the index of the
```
Comment (richvdh) [inline tests/config/test_base.py:87]:
```suggestion
        # Create one template in each directory, whose content is the index of the
```

**Review by richvdh — APPROVED** [review]
lgtm!

**Pull request merged**